frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,10 +1,75 @@
-type Properties = {
-    label: string;
-    type?: 'button' | 'submit';
+import clsx from 'clsx';
+
+import {
+    type ButtonTheme,
+    ButtonSize,
+    IconName,
+    IconSize,
+} from '~/bundles/common/enums/enums';
+import { type ButtonType, type ValueOf } from '~/bundles/common/types/types';
+
+import { Icon } from '../icon/icon';
+import styles from './styles.module.scss';
+
+type ButtonProperties = {
+    label?: string;
+    type?: ButtonType;
+    theme?: ValueOf<typeof ButtonTheme>;
+    size?: ValueOf<typeof ButtonSize>;
+    onClick?: React.MouseEventHandler<HTMLButtonElement>;
+    isDisabled?: boolean;
+    className?: string;
+    isRightIcon?: boolean;
+    isLeftIcon?: boolean;
 };
 
-const Button: React.FC<Properties> = ({ type = 'button', label }) => (
-    <button type={type}>{label}</button>
-);
+const Button: React.FC<ButtonProperties> = ({
```

Comment:
Try to make this component more customizable. 

- Instead of `isRightIcon`, `isLeftIcon` you can pass `RightComponent`, `LeftComponent` props. Some developers adhere to existing JS standards, using words like `Appended`, `Prepended` component
- Instead of `label ` it's better to use `children` prop